### DISCUSSION TITLE
How constructor work ?

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#4790] [General]
I have an entity, I play with constructor to know how it work. But from [this](https://mikro-orm.io/docs/entity-constructors) there some confuse:
<img width="998" alt="image" src="https://github.com/mikro-orm/mikro-orm/assets/38502025/3a9024f9-8995-4a6b-aeb5-f760e144fc8d">

As I understand, title and author will pass to constructor and foo is assigned.
Now I test by my own situation:

```Typescript
@Entity()
export class Permission {
    @PrimaryKey({ type: types.smallint })
    id: number;

    @Property()
    action: string;

    @Property({ nullable: true })
    actionName?: string;

    constructor(actionName: string, action: string) {
        this.actionName = actionName + "12345";
    }
}

 const permissionDtos = [
            {
                "action": "create",
                "actionName": "Create permission"
            },
            {
                "action": "read"
            }
        ]

 const permissions = permissionDtos.map((p) => this.em.create(Permission, p));

```

Now permissions will be

```Typescript
[
            {
                "actionName": "Create permission",
                "action": "create"
            },
            {
                "actionName": "undefined12345",
                "action": "read"
            }
]
```

The point make me confuse is: From doc, it says: "This will extract title and author from the data and pass it to the constructor, and assign only the rest (so here only the foo property) to the created entity". For my situation, it must be: "action and actionName will pass it to constructor", so actionName must be "Create permission12345" instead of "Create permission".
Am I misunderstand something ?

MikroORM version: 5.8.1
NodeJS: 16

**Comment by B4nan** [comment]
~The problem here is that the order of parameters will be the same as order of your properties - and you have this upside down.~

edit: Actually, the real problem here is that the data are used to rehydrate the entity, which overrides the constructor. Will fix that.

**Reply by B4nan** [reply]
This will help: https://github.com/mikro-orm/mikro-orm/commit/8ff3f65907e37f971edd0cc7542d62b7646ec958?w=1
